### PR TITLE
Add missing parameters to transform parameter maps

### PR DESCRIPTION
**Pull request by N-Dekker**

### CODE REVIEW

**Comment by N-Dekker** [comment]
@ViktorvdValk Before addressing https://github.com/InsightSoftwareConsortium/ITKElastix/issues/93 and https://github.com/SuperElastix/elastix/issues/366 I want to clean up the code of those `CreateTransformParametersMap` member functions, with this pull request. Maybe you could have a look, especially at the second commit, "STYLE: Simplified CreateTransformParametersMap overrides" ? The essence is that each `paramsMap->insert` call adds a parameter (parameter name + vector of values). This pull request moves the creation of those parameter values to `ToString` and `ToVectorOfStrings` helper functions.

**Comment by N-Dekker** [comment]
@mstaring @stefanklein Can we possibly discuss this pull request at the sprint, this afternoon?

**Comment by mstaring** [comment]
lgtm

File: Components/Transforms/BSplineDeformableTransformWithDiffusion/elxBSplineTransformWithDiffusion.hxx [hunk 1]
```diff
@@ -1073,6 +1073,29 @@ BSplineTransformWithDiffusion<TElastix>::WriteToFile(const ParametersType & para
...
+auto
+BSplineTransformWithDiffusion<TElastix>::CreateDerivedTransformParametersMap(void) const -> ParameterMapType
+{
+  const auto & itkTransform = *m_BSplineTransform;
+  const auto   gridRegion = itkTransform.GetGridRegion();
+
+  // TODO If necessary, add possibly missing parameters:
+  // - "DeformationFieldFileName" (which is written by WriteToFile).
+  // - "GridDirection" (as returned by RecursiveBSplineTransform).
+  // - "ThresholdBool", "ThresholdHU", etc. (as retrieved by ReadParameter).
```
Comment (N-Dekker) [inline Components/Transforms/BSplineDeformableTransformWithDiffusion/elxBSplineTransformWithDiffusion.hxx:1090]: Note, still a TODO here...

File: Components/Transforms/MultiBSplineTransformWithNormal/elxMultiBSplineTransformWithNormal.hxx [hunk 2]
```diff
@@ -705,6 +705,29 @@ MultiBSplineTransformWithNormal<TElastix>::WriteToFile(const ParametersType & pa
...
+
+template <class TElastix>
+auto
+MultiBSplineTransformWithNormal<TElastix>::CreateDerivedTransformParametersMap(void) const -> ParameterMapType
+{
+  const auto & itkTransform = *m_MultiBSplineTransformWithNormal;
+  const auto   gridRegion = itkTransform.GetGridRegion();
+
+  // TODO If necessary, add possibly missing parameters:
+  // - "MultiBSplineTransformWithNormalLabels" (which is written by WriteToFile).
```
Comment (N-Dekker) [inline Components/Transforms/MultiBSplineTransformWithNormal/elxMultiBSplineTransformWithNormal.hxx:720]: ...and another TODO here!

**Pull request merged**